scripts/ensure-bundle-deps.ts

```diff
@@ -0,0 +1,146 @@
+  //   dependencies - but it's helpful to consider all the possibilities.)
+  //
+  // In the first version of the first scenario (repo-level `build` -> repo-level `build:bundle`), all we have to do is
+  // wait, because other parts of `build` are creating them as this check is being done. (Waiting 5 or 10 or even 15
+  // seconds to start running `build:bundle` in parallel is better than pushing it to the second half of `build`,
+  // because `build:bundle` is the slowest part of the build and therefore the one we most want to parallelize with
+  // other slow parts, like `build:types`.)
+  //
+  // In all other scenarios, if the dependencies are missing, we have to build them ourselves - with `build:bundle` at
+  // either level, we're the only thing happening (so no one's going to do it for us), and with package-level `build`,
+  // types and npm assets are being built simultaneously, but only for the package being bundled, not for its
+  // dependencies. Either way, it's on us to fix the problem.
+  //
+  // TODO: This actually *doesn't* work for package-level `build`, not because of a flaw in this logic, but because
+  // `build:rollup` has similar dependency needs (it needs types rather than npm builds). We should do something like
+  // this for that at some point.
+
+  if (isTopLevelBuild && yarnScript === 'build') {
+    let retries = 0;
+
+    console.log('\nSearching for bundle dependencies...');
+
+    while (retries < maxRetries && !checkForBundleDeps(packagesDir, dependencyDirs)) {
+      console.log('Bundle dependencies not found. Trying again in 5 seconds.');
+      retries += 1;
+      await sleep(5000);
+    }
+
+    if (retries === maxRetries) {
+      throw new Error(
+        `\nERROR: \`yarn build:bundle\` (triggered by \`yarn build\`) cannot find its depdendencies, despite waiting ${
+          5 * maxRetries
+        } seconds for the rest of \`yarn build\` to create them. Something is wrong - it shouldn't take that long. Exiting.`,
+      );
+    }
+
+    console.log(`\nFound all bundle dependencies after ${retries} retries. Beginning bundle build...`);
+  }
+
+  // top-level `build:bundle`, package-level `build` and `build:bundle`
+  else {
+    console.warn('\nWARNING: Missing dependencies for bundle build. They will be built before continuing.');
+
+    for (const dependencyDir of dependencyDirs) {
+      console.log(`\nBuilding \`${dependencyDir}\` package...`);
+      run('yarn build:dev', { cwd: `${packagesDir}/${dependencyDir}` });
+    }
+
+    console.log('\nAll dependencies built successfully. Beginning bundle build...');
+  }
+}
+
+/**
+ * See if all of the necessary dependencies exist
+ */
+function checkForBundleDeps(packagesDir: string, dependencyDirs: string[]): boolean {
+  for (const dependencyDir of dependencyDirs) {
+    const depBuildDir = `${packagesDir}/${dependencyDir}/build`;
+    // const depBuildDir = path.resolve(process.cwd(), packagesDir, dependencyDir, 'build');
```

Comment:
```suggestion
```

Reply:
Damn, I thought I got rid of that. Must have been in a different version of the branch. Thanks for catching it.